library/std/src/thread/mod.rs

```diff
@@ -1445,8 +1445,15 @@ impl<T> JoinHandle<T> {
 
     /// Checks if the associated thread is still running its main function.
     ///
+    /// `is_running` supports implementing a non-blocking join operation, by
+    /// checking `is_running`, and calling `join` if it returns `false`.
+    /// `is_running` is not intended to be tightly polled in a busyloop or
+    /// similar; use `join` to block waiting on another thread.
+    ///
     /// This might return `false` for a brief moment after the thread's main
     /// function has returned, but before the thread itself has stopped running.
+    /// However, if this returns `false`, `join` can be expected to return
+    /// relatively quickly, and not block for any significant amount of time.
```

Comment:
Just a thought: we could 'fix' this by just always `pthread_detach`ing every spawned thread directly, and only using our own return value `Arc` for 'join'ing threads. (This is also how I implemented scoped threads.) Then we can make sure that `join` will instantly return with no possibility of blocking as soon as the `Arc` is ready.